### DISCUSSION TITLE
Horus X12S externalaccessmod compatibility

### DISCUSSION THREAD

**Discussion by Unb0rn** [EdgeTX/edgetx#2326] [Q&A]
I have an extensively modified Horus X12S (ISRM module, Para Bluetooth, touchscreen, fast inverter, external access), I've been waiting for 2.8 release to have the touchscreen enabled in the official build but got a bit impatient and built it myself. The problem is - my R9M 2019 ACCESS (Flex) no longer works...
I lost my previous build but managed to find one in [this](https://github.com/EdgeTX/edgetx/discussions/1951) discussion, reflashed it and it looks like access works fine in 2.7.1 but is not detected in own-built firmware from master branch - the option is there but the module is not detected.
I feel a bit lost - are there any new requirements for this mod? Has anyone else tried this configuration?

**Comment by raphaelcoeffic** [comment]
Regarding external ACCESS, it works for me on a modded T16. If the compilation options are correct, it should work.

Now the question is, what in your particular case does the module prevent from working properly.

As it seems to work for you on 2.7.1 (with the same compilation options?), the question would be, what exactly changed here. I must admit it is hard to debug remotely.

**Reply by Unb0rn** [reply]
Well, I can't remember what my options were for 2.7 build, but I'm pretty sure it was just `HARDWARE_EXTERNAL_ACCESS_MOD=YES`
I also found some other builds for X12S in [this](https://github.com/EdgeTX/edgetx/pull/2211#issuecomment-1225497214) PR, which doesn't work for me too... 
So it feels more like 2.7 without touchscreen - everything works fine, 2.8 with touchscreen working - externalaccessmod cannot find the module...

**Reply by raphaelcoeffic** [reply]
If you could do a bisection to find the culprit commit it might help us to find the issue.

**Reply by Unb0rn** [reply]
I had no time to do a normal bisect, but I can confirm that:
- externalaccessmod works fine on latest commit on 2.7 branch (0713f10)
- externalaccessmod works fine with touchscreen display mod on main branch when the touchscreen was added (7a6d623) <Apr 23>
- externalaccessmod works fine on commit 162880b <Jun 12>
- externalaccessmod does NOT work on the latest main commit (40149c4) <Sep 22>
- it was not working like a week ago
So the problem is somewhere between June 12th and September 16th...

**Reply by Unb0rn** [reply]
Made a couple more builds and narrowed the problem down to 3 commits - it turns out the problem is somewhere between commits ab6033e (working fine) and db73e4c (broken).
Most probably its this one: f952c5f7
@raphaelcoeffic do I have to create an issue?

**Reply by pfeerick** [reply]
All three of those commits are from https://github.com/EdgeTX/edgetx/pull/2100, so it's no real surprise to find it was the cuplrit... it seems to have been the PR that keeps on giving! 😆

**Reply by raphaelcoeffic** [reply]
So possibly related to internal module RX DMA? If this is the case, I should be able to test this with my T16-ISRM.

**Reply by raphaelcoeffic** [reply]
@Unb0rn i believe this is fixed and should work with 2.8.0-RC3. I tested it several times lately.
